pkgs/top-level/all-packages.nix

```diff
@@ -357,6 +357,13 @@ with pkgs;
 
   mumps_par = callPackage ../by-name/mu/mumps/package.nix { mpiSupport = true; };
 
+  prism-model-checker-unwrapped = callPackage ../by-name/pr/prism-model-checker-unwrapped/package.nix {
+      java = openjdk17;
+      # makefile hardcodes gcc for darwin
+      stdenv = if stdenv.hostPlatform.isDarwin then gccStdenv else stdenv;
+  };
+  prism-model-checker = callPackage ../by-name/pr/prism-model-checker/package.nix { java = openjdk17; };
+
```

Comment:
Avoid adding new stuff in `all-packages.nix`. Just import `openjdk17` in both derivs and  move the stdenv logic into a let-in in `-unwrapped`